okta/okta-auth-js
# Test app: handle !isAuthenticated if document is visible

New behavior on change of `authState.isAuthenticated` to false for `/protected` page: 
- call `signInWithRedirect` only if document is visible and it's app initialisation stage
- otherwise show message `You are NOT authenticated. Sign-in again using one of these methods: [sign-in buttons]`

Internal ref: [OKTA-456247](https://oktainc.atlassian.net/browse/OKTA-456247)

---

Steps for `Test auth required`:
- Open protected page in new tab 1 and trigger logout
 Expected text in tab 1: `You are NOT authenticated. Sign-in again using one of these methods: [sign-in buttons]`
- Current tab has visibility state hidden for now (not in focus because of opened tab 1).
Expected text in current tab: `You are authenticated`
- Close tab 1 and return visibility for the current tab.
Then expected text in current tab: `You are NOT authenticated. Sign-in again using one of these methods: [sign-in buttons]`
- Open tab 2
Expected text in tab 2: `You are NOT authenticated. You are being redirected to sign-in page automatically...`
- Close tab 2


https://user-images.githubusercontent.com/72614880/157540431-0fe75bfe-d9fd-4150-bfa9-34066d2520eb.mov
